lib/step.js

```diff
@@ -112,10 +112,10 @@ class Step {
         return `${arg}`;
       } else if (arg instanceof Secret) {
         return '*****';
-      } else if (arg.toString && arg.toString() !== '[object Object]') {
-        return arg.toString();
       } else if (typeof arg === 'object') {
         return JSON.stringify(arg);
+      } else if (arg.toString && arg.toString() !== '[object Object]') {
```

Comment:
You can checks that arg is not null (undefined) at the beginning. 
`arg && arg.toString && arg.toString() !== '[object Object]'`
And not rearrange conditions.

Reply:
@Vorobeyko pls see my comment below